pkg/capture/capture_mock_test.go

```diff
@@ -0,0 +1,108 @@
+package capture
+
+import (
+	"net"
+	"time"
+
+	"github.com/fako1024/slimcap/capture"
+	"github.com/fako1024/slimcap/event"
+	"github.com/fako1024/slimcap/link"
+)
+
+type mockCapture struct {
+	nPkts, maxPkts int
+	block          chan event.EvtData
+}
+
+func newMockCaptureSource(maxPkts int) *mockCapture {
+	return &mockCapture{
+		maxPkts: maxPkts,
+		block:   make(chan event.EvtData, 1),
+	}
+}
+
+// NewPacket creates an empty "buffer" package to be used as destination for the NextPacketInto()
+// method. It ensures that a valid packet of appropriate structure / length is created
+func (c *mockCapture) NewPacket() capture.Packet {
+	panic("not implemented") // TODO: Implement
```

Comment:
Now, I'm slowly making the connection to https://github.com/fako1024/slimcap/pull/14#pullrequestreview-1357368146 😄

Reply:
Jup. When I hacked this together I immediately thought that this should be generalized / made available in `slimcap`. :rofl: